tests/karma/unit/controllers/inbox.js

```diff
@@ -0,0 +1,195 @@
+        return {
+          init: sinon.stub()
+        };
+      });
+      $provide.factory('DeleteDoc', function() {
+        spyDeleteDoc = sinon.spy();
+        return spyDeleteDoc;
+      });
+      $provide.factory('DownloadUrl', function() {
+        return sinon.stub();
+      });
+      $provide.factory('Enketo', function() {
+        return {
+          withAllForms: function() {
+            return Promise.resolve({});
+          }
+        };
+      });
+      $provide.factory('Facility', function() {
+        return sinon.stub();
+      });
+      $provide.factory('FacilityHierarchy', function() {
+        return sinon.stub();
+      });
+      $provide.factory('Form', function() {
+        return sinon.stub();
+      });
+      $provide.factory('Language', function() {
+        return function() {
+          return Promise.resolve({});
+        };
+      });
+      $provide.factory('ReadMessages', function() {
+        return sinon.stub();
+      });
+      $provide.factory('SendMessage', function() {
+        return sinon.stub();
+      });
+      $provide.factory('Session', function() {
+        return {
+          init: sinon.stub()
+        };
+      });
+      $provide.factory('SetLanguageCookie', function() {
+        return sinon.stub();
+      });
+      $provide.factory('Settings', function() {
+        return sinon.stub();
+      });
+      $provide.factory('$state', function() {
+        spyState = {
+          go: sinon.spy()
+        };
+        return spyState;
+      });
+      $provide.factory('$stateParams', function() {
+        return sinon.stub();
+      });
+      $provide.factory('$timeout', function() {
+        return sinon.stub();
+      });
+      $provide.factory('translateFilter', function() {
+        return sinon.stub();
+      });
+      $provide.factory('UpdateUser', function() {
+        return sinon.stub();
+      });
+      $provide.factory('UpdateSettings', function() {
+        return sinon.stub();
+      });
+      $provide.factory('UserDistrict', function() {
+        return sinon.stub();
+      });
+      $provide.factory('UserSettings', function() {
+        return sinon.stub();
+      });
+      $provide.factory('$window', function() {
+        return sinon.stub();
+      });
+      $provide.constant('APP_CONFIG', {
+        name: 'name',
+        version: 'version'
+      });
+    });
+
+    inject(function($rootScope, $controller) {
+      scope = $rootScope.$new();
+      scope.fetchAnalyticsModules = function() {
+        return Promise.resolve({});
+      };
+      createController = function() {
+        return $controller('InboxCtrl', {
+          '$scope': scope,
+          '$rootScope': $rootScope
+        });
+      };
+    });
+  });
+
+  afterEach(function() {});
+
+  it('creates controller', function() {
+    createController();
+  });
+
+  it('deletes contact', function() {
+    createController();
+    spyDeleteDoc.reset();
```

Comment:
Is it possible to reset all the spies at the end of each test, instead of having to manually do it inside each test?  Or is this the preferred style?